salesforce/bulkwrite.go

```diff
@@ -374,10 +374,6 @@ func (c *Connector) getPartialFailureDetails(ctx context.Context, jobInfo *GetJo
 			failureMap = failInfo.FailedCreates
 		}
 
-		if failureMap == nil {
```

Comment:
This `failureMap` is never `nil` because we set it up in line 367 as `failInfo.FailedUpdates` and `failInfo.FailedUpdates` is not `nil` according to assignemtn in line 330. 

`failure[errMsg] == nil` check is not necessary since `failureMap[errMsg] = append(failureMap[errMsg], externalId)` will append without resulting in nil pointer exception for `failureMap[errMsg]` is typed `nil`